db/db_test.go

```diff
@@ -251,6 +242,125 @@ func Test_DB_Exist(t *testing.T) {
 	}
 }
 
+func Test_DB_StoreLastProcessedBlock(t *testing.T) {
+	testTable := []struct {
+		name      string
+		task      string
+		block     uint64
+		returnErr error
+	}{
+		{
+			name:  "value inserted",
+			task:  "task1",
+			block: 1,
+		},
+		{
+			name:      "error returned",
+			task:      "task1",
+			block:     1,
+			returnErr: errors.New("test error"),
+		},
+	}
+
+	for _, tt := range testTable {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			db, mock, err := sqlmock.New()
+			require.NoError(t, err)
+
+			defer db.Close()
+
+			expected := mock.ExpectExec(`INSERT INTO data_node\.sync_tasks \(task, block\) VALUES \(\$1, \$2\) ON CONFLICT \(task\) DO UPDATE SET block = EXCLUDED\.block, processed = NOW\(\)`).
+				WithArgs(tt.task, tt.block)
+			if tt.returnErr != nil {
+				expected.WillReturnError(tt.returnErr)
+			} else {
+				expected.WillReturnResult(sqlmock.NewResult(1, 1))
+			}
+
+			wdb := sqlx.NewDb(db, "postgres")
+
+			dbPG := New(wdb)
+
+			err = dbPG.StoreLastProcessedBlock(context.Background(), tt.task, tt.block, wdb)
+			if tt.returnErr != nil {
+				require.ErrorIs(t, err, tt.returnErr)
+			} else {
+				require.NoError(t, err)
+			}
+
+			require.NoError(t, mock.ExpectationsWereMet())
+		})
+	}
+}
+
+func Test_DB_GetLastProcessedBlock(t *testing.T) {
+	testTable := []struct {
+		name      string
+		task      string
+		block     uint64
+		returnErr error
+	}{
+		{
+			name:  "successfully selected block",
+			task:  "task1",
+			block: 1,
+		},
+		{
+			name:      "error returned",
+			task:      "task1",
+			block:     1,
+			returnErr: errors.New("test error"),
+		},
+	}
+
+	for _, tt := range testTable {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			db, mock, err := sqlmock.New()
+			require.NoError(t, err)
+
+			defer db.Close()
+
+			mock.ExpectExec(`INSERT INTO data_node\.sync_tasks \(task, block\) VALUES \(\$1, \$2\) ON CONFLICT \(task\) DO UPDATE SET block = EXCLUDED\.block, processed = NOW\(\)`).
```

Comment:
Can we use those constants for sql queries in `db.go` here in the mocks? Like, export them in the db package, and then use them here in the `ExpectExec` functions.

Reply:
In this case, those test cases are going to be useless. Here in tests we explicitly define how SQL query should look like so if there gonna be a mistake in the query, current tests will catch it. In case when we use a constant in both implementation and tests, a potential query mistake would not be caught.